src/frontend/src/utils/iiConnection.ts

```diff
@@ -229,20 +233,68 @@ export class Connection {
 
     const actor = await this.createActor(delegationIdentity);
 
+    const connection = new AuthenticatedConnection(
+      this.canisterId,
+      identity,
+      delegationIdentity,
+      userNumber,
+      actor
+    );
+
+    const attachmentInfo = identity.getAuthenticatorAttachment();
+    if (attachmentInfo !== undefined) {
+      this.updateKeyTypeIfNecessary(devices, attachmentInfo, connection);
+    }
+
     return {
       kind: "loginSuccess",
       userNumber,
-      connection: new AuthenticatedConnection(
-        this.canisterId,
-        // eslint-disable-next-line
-        identity,
-        delegationIdentity,
-        userNumber,
-        actor
-      ),
+      connection,
     };
   };
 
+  private updateKeyTypeIfNecessary(
```

Comment:
I'm not sure what this does. You want to ask @nmattia to review as well? 

Can rubber stamp if you want.

Reply:
I'll ask @nathanosdev, since he's my review buddy while @nmattia is away.
No need to rubber stamp.